### PR TITLE
feat: rich breadcrumbs

### PR DESCRIPTION
**Pull request by thiskevinwang**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](https://app.asana.com/0/0/1202888076877382/f) 🎟️

## 🗒️ What

This updates our `BreadcrumbBar` to implement the schema for rich results, as documented in https://developers.google.com/search/docs/advanced/structured-data/breadcrumb#breadcrumb-list

## 📸 Design Screenshots

No visual change

## 🧪 Testing

https://search.google.com/test/rich-results

### CODE REVIEW

File: src/components/breadcrumb-bar/index.tsx [hunk 1]
```diff
@@ -34,35 +35,71 @@ function BreadcrumbBar({
...
+				<script
+					type="application/ld+json"
+					dangerouslySetInnerHTML={{ __html: stringifiedStructuredData }}
+				/>
+			</Head>
 			<ol className={s.listRoot}>
-				{links.map(({ title, url, isCurrentPage }) => {
-					const cleanTitle = title.replace(/<[^>]+>/g, '')
-					const Elem = url ? InternalLink : 'span'
+				{links.map(({ title, url, isCurrentPage }, i) => {
```
Comment (ashleemboyer) [inline src/components/breadcrumb-bar/index.tsx:63]: Doesn't appear that `i` is used here now.

```suggestion
				{links.map(({ title, url, isCurrentPage }) => {
```

File: src/components/breadcrumb-bar/index.tsx [hunk 2]
```diff
@@ -34,35 +35,71 @@ function BreadcrumbBar({
...
+			<Elem
+				className={s.breadcrumbText}
+				href={url}
+				aria-current={isCurrentPage ? 'page' : undefined}
+			>
+				<span itemProp="name">{cleanTitle}</span>
+			</Elem>
+		</Text>
+	)
+}
```
Comment (ashleemboyer) [inline src/components/breadcrumb-bar/index.tsx:101]: A couple of questions/notes here:

- What is the purpose of these changes overall?
- Do `Breadcrumb` and `BreadCrumbProps` need to be exported? I'm not sure we'll want these to be useable outside of this component.
- The `itemProp` on the `span` isn't needed, so I think that line can be changed back to just `{cleanTitle}`.

**Review by ashleemboyer — CHANGES_REQUESTED** [review]

File: src/components/breadcrumb-bar/index.tsx (hunk 2 above)
Comment (thiskevinwang) [reply src/components/breadcrumb-bar/index.tsx:101]: I previously exported these new components for a local test, but that file has since been deleted.

I'll go and revert these changes.

File: src/components/breadcrumb-bar/index.tsx [hunk 3]
```diff
@@ -34,35 +35,71 @@ function BreadcrumbBar({
...
+	const structuredData = [
+		{
+			'@context': 'https://schema.org',
+			'@type': 'BreadcrumbList',
+			itemListElement: links.map((link, index) => ({
+				'@type': 'ListItem',
+				position: index + 1,
+				name: link.title,
+				item: link.url
+					? `https://developer.hashicorp.com${link.url}`
```
[the remark this answers is not in the document]
Comment (thiskevinwang) [reply src/components/breadcrumb-bar/index.tsx:48]: I don't think so and reasoning is: this _structuredData_ will be server-rendered, and `window` won't be available, plus this _structuredData_ won't be relevant for non-production deployments.

Certainly open to other thoughts here!

[the remark this answers is not in the document]
Comment (ashleemboyer) [reply src/components/breadcrumb-bar/index.tsx:48]: Got it! That makes sense to me.

**Review by ashleemboyer — APPROVED** [review]
Looks great to me! 🙌🏻  Here are the test results I got from the preview URL, I believe this checks out!

image
